streams/delimiter_stream.ts

```diff
@@ -195,8 +195,6 @@ export class DelimiterStream extends TransformStream<Uint8Array, Uint8Array> {
             chunkStart = disposition === "prefix"
               ? delimitedChunkEnd
               : inspectIndex;
-          } else {
-            throw new Error("unreachable");
```

Comment:
I prefer to keep this line. If the user his this line, this error helps us debugging the issue.

Reply:
Reverted for now, but created #5090 for further investigation.